### ISSUE TITLE
[Feature] Integrating Godot 4.3 Motion Matching for Humanoid Orientation Prediction

### ISSUE THREAD

**Issue by GeorgeS2019** [GodotVR/godot-xr-tools#674]
https://github.com/GuilhermeGSousa/godot-motion-matching/discussions/30

**Comment by DigitalN8m4r3** [comment]
while this certainly is interesting, i dont think that its something that should be part of xr tools and neither in the v2 Version of it. We currently still have other areas that need attention way more then something like this. There is avatar work already done by malcolm and it is in a verry useable state.

**Comment by GeorgeS2019** [comment]
https://github.com/GuilhermeGSousa/godot-motion-matching/discussions/30#discussioncomment-10595993

**Comment by Malcolmnixon** [comment]
> while this certainly is interesting, i dont think that its something that should be part of xr tools and neither in the v2 Version of it. We currently still have other areas that need attention way more then something like this. There is avatar work already done by malcolm and it is in a verry useable state.

I agree - my current work is all in smaller assets that can cooperatively work together - that way developers don't end up with a massive monolithic asset where they only use 10% of it.

That being said I thing a Godot Motion Matching asset would be extremely valuable.

**Comment by BastiaanOlij** [comment]
Motion matching is super cool and I definately hope more Godot plugins will be created to offer solutions for this.

However this is not a core issue to solve in XR Tools.

**Comment by GeorgeS2019** [comment]
@BastiaanOlij @Malcolmnixon 

[Godot cesium is working](https://github.com/Battle-Road-Labs/3D-Tiles-For-Godot/issues/84#issuecomment-2888733987), we need to create **an XR environment in Godot cesium** where **a user** or **many AI agents can automatically navigate with[ motion matching](https://github.com/GuilhermeGSousa/godot-motion-matching/issues/98)**. 

We are there NOW. We just need coordination in getting there.

**Comment by BastiaanOlij** [comment]
@GeorgeS2019 totally agree that having systems like this would be amazing for those who need it. But this is far out of scope for what XR Tools is meant to be and this is not the right place to discuss this.

To be exact, this is one of the **big** reasons Malcolm and myself are working on an XR Tools Lite alternative. XR Tools is already in a state where it is trying to do soo many different things its becoming unwieldy, and with XR Tools 2 taking a focus on pure physics interaction it's become very apparent that this is just leaving a lot of people behind.

There is no once size fits all solution in XR. Building what you're hinting at here is going to be the best thing since sliced bread for a portion of the XR community, and a total drag on the rest of us. We need far more modular solutions where people can add in the pieces they need and not be held back by solutions only a few people need.

**Comment by GeorgeS2019** [comment]
Thx for updating where XR team is thinking and how to modularise the development
